Automattic/wp-calypso
# Google My Business: Segment Track events by type of site

This pull request introduces a way in our stats to segment users who access Google My Business pages according to the type of their site (i.e. WordPress.com or Jetpack). More specifically, it updates all Track events triggered from those pages with a `site_type` prop using a new `withEnhancers()` helper function that can be employed to modify a Redux action before dispatching it.

This pull request takes advantage of this new helper to remove duplicate code that was used in `handleConnect()` event handlers. It also makes other event handler names more consistent with event sources.
  
#### Testing instructions
 
1. Run `git checkout fix/google-my-business-jetpack-stats` and start your server, or open a [live branch](https://calypso.live/?branch=fix/google-my-business-jetpack-stats)
2. Open the [`Stats` page](http://calypso.localhost:3000/stats) of a WordPress.com site that is eligible to Google My Business
3. Enter `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` in your browser's console
4. Reload the page, and click the `Start Now` button in the nudge
5. Proceed to the `Stats` page while checking that all actions fire a Tracks event
6. Check that a `dismiss_count` prop is added to events fired by the nudge
7. Check that the `location_count` and `verified_location_count` props are added to events fired by the connection process
8. Check that all Tracks events have a `site_type` prop set to `wpcom`
9. Repeat step #2 to #7 with a Jetpack site this time
10. Check that all Tracks events have a `site_type` prop set to `jetpack`

#### Reviews
